### PR TITLE
Tweaking cmd ctrl refactor

### PR DESCRIPTION
**Pull request by oetyng**
Based on PR #1402, but changes the periodic checks to run on an interval, instead of based on when there are incoming connections.

### CODE REVIEW

File: sn_node/src/node/flow_ctrl/cmd_ctrl.rs [hunk 1]
```diff
@@ -280,7 +255,7 @@ impl std::hash::Hash for EnqueuedJob {
     fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
         //self.job.hash(state);
         self.job.id().hash(state);
-        self.retries.hash(state);
+        // self.retries.hash(state);
```
Comment (davidrusu) [inline sn_node/src/node/flow_ctrl/cmd_ctrl.rs:258]:
```suggestion
```

File: sn_node/src/node/flow_ctrl/cmd_ctrl.rs [hunk 2]
```diff
@@ -264,7 +239,7 @@ impl MsgThroughput {
 #[derive(Debug)]
 pub(crate) struct EnqueuedJob {
     job: CmdJob,
-    retries: usize,
+    // retries: usize,
```
Comment (davidrusu) [inline sn_node/src/node/flow_ctrl/cmd_ctrl.rs:242]:
```suggestion
```

File: sn_node/src/node/flow_ctrl/mod.rs [hunk 3]
```diff
@@ -47,30 +50,152 @@ const DYSFUNCTION_CHECK_INTERVAL: Duration = Duration::from_secs(5);
...
+        let mut last_data_batch_check = Instant::now();
+        let mut last_link_cleanup = Instant::now();
+        let mut last_dysfunction_check = Instant::now();
         #[cfg(feature = "back-pressure")]
-        ctrl.clone().start_backpressure_reporting();
+        let mut last_backpressure_check = Instant::now();
+
+        // the internal process loop
+        loop {
+            debug!("Internal looop");
```
Comment (davidrusu) [inline sn_node/src/node/flow_ctrl/mod.rs:88]: Do we sleep ever? this looks like it will busy loop

**Comment by davidrusu** [comment]
Ah missed the `DO NOT MERGE` tag, apologies for the early review

**Comment by joshuef** [comment]
Closed as this was incorporated into the original PR #1402

**Pull request closed without merging**